src/lib/server/product.ts

```diff
@@ -155,3 +159,34 @@ export async function amountOfProductSold(productId: string): Promise<number> {
 		)?.total ?? 0
 	);
 }
+
+export function productPriceWithVariations(
+	product: Product,
+	chosenVariations: Record<string, string> | undefined
+) {
+	let variationPriceArray: Price[] = [];
+	const variationNamesInDB = [...new Set(product.variations?.map((vari) => vari.name))];
+	const chosenVariationNames = Object.keys(chosenVariations ?? {});
+
+	const allVariationsChosen =
+		variationNamesInDB.length === chosenVariationNames.length &&
+		variationNamesInDB.every((name) => chosenVariationNames.includes(name));
+
+	if (allVariationsChosen) {
```

Comment:
Can you move this to a separate function

So we have two functions:

- `checkProductVariationsIntegrity` here in `src/lib/server/product.ts`
- `productPriceWithVariations` in `src/lib/types/Product.ts`

So then we can reuse `productPriceWithVariations` in the frontend - in `product/[id]/+page.svelte`